bin/propolis-cli/src/main.rs

```diff
@@ -219,14 +248,22 @@ async fn new_instance(
     cloud_init_bytes: Option<String>,
     silo_id: TypedUuid<SiloKind>,
     project_id: TypedUuid<ProjectKind>,
+    sled_id: TypedUuid<SledKind>,
+    sled_model: String,
+    sled_revision: u32,
+    sled_serial: String,
```

Comment:
It's time to start wrapping some of these args in structs, now that we're at a baker's dozen

Reply:
Done in 8a71e3d